packages/entity-list/src/util/preferences.js

```diff
@@ -39,3 +39,42 @@ export const getPositionsPreferencesToSave = (formName, positions) =>
       ...acc,
       [`${formName}.${key}.position`]: positions[key].toString()
     }), {})
+
+export const getSorting = preferences => {
+  const keyRegex = /^.*\.sorting(Field|Direction)\.?([1-9]*)$/
+  return Object.entries(preferences)
+    .filter(([key]) => keyRegex.test(key))
+    .map(([key, value]) => [transformSortingKey(key.match(keyRegex)), value])
+    .reduce((acc, [key, value]) => {
+      if (!acc[key.index]) {
+        acc[key.index] = {}
+      }
+      acc[key.index][key.field] = value
+      return acc
+    }, [])
+}
+
+const transformSortingKey = keyMatch => {
+  return {
+    field: transformSortingField(keyMatch[1]),
+    index: keyMatch[2] ? parseInt(keyMatch[2]) : 0
+  }
+}
+
+function transformSortingField(field) {
```

Comment:
Any reason why this is a named function?
Anyhow, this is shorter:

`const transformSortingField = field => field === 'Direction' ? 'order' :  field.toLowerCase()`

Reply:
Because my IDE still isn't configured to not automatically generate these when I extract code to a method :P